src/wp-includes/class-wp-theme-json.php

```diff
@@ -2901,9 +2901,9 @@ public function get_root_layout_rules( $selector, $block_metadata ) {
 			// Alignfull children of the container with left and right padding have negative margins so they can still be full width.
 			$css .= '.has-global-padding > .alignfull { margin-right: calc(var(--wp--style--root--padding-right) * -1); margin-left: calc(var(--wp--style--root--padding-left) * -1); }';
 			// Nested children of the container with left and right padding that are not wide or full aligned do not get padding.
-			$css .= '.has-global-padding :where(.has-global-padding:not(.wp-block-block, .alignfull, .alignwide)) { padding-right: 0; padding-left: 0; }';
+			$css .= '.has-global-padding :where(:not(.alignfull.is-layout-flow) > .has-global-padding:not(.wp-block-block, .alignfull, .alignwide)) { padding-right: 0; padding-left: 0; }';
 			// Nested children of the container with left and right padding that are not wide or full aligned do not get negative margin applied.
```

Comment:
```suggestion
			// Alignfull direct children of the containers that are targeted by the rule above do not need negative margins.
```

This comment was updated for accuracy and clarity in the Gutenberg PR as well.